packages/@ember/runloop/index.js

```diff
@@ -217,7 +217,39 @@ export function join() {
   @since 1.4.0
   @public
 */
-export const bind = (...curried) => (...args) => join(...curried.concat(args));
+export const bind = (...curried) => {
+  if (!bindMethodPresent(...curried)) {
+    throw new Error('could not find a suitable method to bind');
```

Comment:
Can we make this an assertion and inline the `bindMethodPresent` implementation in an IIFE as the predicate?

```js
assert('Could not find a suitable method to bind.', (function() {
  // guts of conditional below...
})());
```